src/Cli/dotnet/commands/dotnet-run/Program.cs

```diff
@@ -61,8 +61,9 @@ public static RunCommand FromParseResult(ParseResult parseResult)
                 noRestore: parseResult.HasOption(RunCommandParser.NoRestoreOption) || parseResult.HasOption(RunCommandParser.NoBuildOption),
                 interactive: parseResult.HasOption(RunCommandParser.InteractiveOption),
                 verbosity: parseResult.HasOption(CommonOptions.VerbosityOption) ? parseResult.GetValue(CommonOptions.VerbosityOption) : null,
-                restoreArgs: restoreArgs.ToArray(),
-                args: nonBinLogArgs.ToArray()
+                restoreArgs: [.. restoreArgs],
+                args: [.. nonBinLogArgs],
+                environmentVariables: [.. CommonOptions.GetEnvironmentVariables(parseResult)]
```

Comment:
I think you could push this parsing into the [CliOption's definition directly](https://github.com/dotnet/sdk/blob/0ee30b8bee52e1104e5532645397a7450a226cde/src/Containers/containerize/ContainerizeCommand.cs#L300-L319), making the option easier to use.